### ISSUE TITLE
Animation onComplete callback

### ISSUE THREAD

**Issue by marioconforti** [mdewilde/chart#26]
I'm trying to use an animation onComplete js function, in this way:

```
Animation<?> animation = new Animation<>();
		JavaScriptFunction js = new JavaScriptFunction("function(){alert(chart.toBase64Image())}");
		animation.setOnComplete(js);
		options.setAnimation(animation);
```

this is rendered in this way inside options

```
"animation" : {
      "onComplete" : function(animation){console.log(chart.toBase64Image())}
    },
```
but the graph is not rendered at all, and i have no logs to see what's going on.
If i put off the option, the graph is rendered correctly.

**Comment by marioconforti** [comment]
the problem should be that 
```
"animation" : {
      "onComplete" : function(){console.log(chart.toBase64Image())}
    }
``` is not valid JSON and adding "" will not call the javascript function

Charts.js is expecting valid json when you use it? Because if i pass it as a string.

```
{
  "type" : "line",
  "data" : {
    "datasets" : [ {
      "data" : [ {
        "t" : 1551273300000,
        "y" : 20.75
      }, {
        "t" : 1551273600000,
        "y" : 20.75
      }, {
        "t" : 1551273900000,
        "y" : 20.75
      }, {
        "t" : 1551274200000,
        "y" : 20.75
      }, {
        "t" : 1551274500000,
        "y" : 20.75
      }, {
        "t" : 1551274800000,
        "y" : 20.75
      }, {
        "t" : 1551275100000,
        "y" : 20.75
      }, {
        "t" : 1551275400000,
        "y" : 20.75
      },  {
        "t" : 1551277800000,
        "y" : 28.0625
      } ],
      "label" : "Label",
      "fill" : false,
      "backgroundColor" : "rgba(0,128,0,1.000)",
      "borderWidth" : 2,
      "borderColor" : "rgba(0,128,0,1.000)"
    } ]
  },
  "options" : {
    "responsive" : true,
    "maintainAspectRatio" : true,
    "title" : {
      "display" : true,
      "position" : "top",
      "text" : "aaaaa"
    },
    "legend" : {
      "position" : "bottom"
    },
    "hover" : {
      "mode" : "dataset"
    },
    "animation" : {
      "onComplete" : function(animation){alert("ok")}
    },
    "scales" : {
      "xAxes" : [ {
        "type" : "time",
        "time" : {
          "displayFormats" : {
            "millisecond" : null,
            "second" : null,
            "minute" : "HH:mm",
            "hour" : "DD/MM HH:mm",
            "day" : "DD/MM HH:mm",
            "week" : null,
            "month" : "DD/MM HH",
            "quarter" : null,
            "year" : null
          },
          "tooltipFormat" : "DD/MM/YY HH:mm"
        }
      } ]
    },
    "elements" : {
      "point" : {
        "radius" : 1,
        "hitRadius" : 2,
        "hoverRadius" : 2
      }
    }
  }
}
```

i got an error in JS -> `Cannot create property 'data' on string`

**Comment by dan-whitehouse** [comment]
Yeah, I just recently downloaded the library and ran into the same issue, just a different callback. My workaround was to  exclude the callbacks from the ChartOptions and just include it in my javascript.

`        <script>

        const submissionsTodayByRole = (function() {
        const $chart = $('#chart-submissions-today');

        let $fetch = function () {
          $.ajax({
            url: '/v3/api/dashboard/responsesByRole',
            method: 'GET',
            dataType: "json",
            contentType: "application/json",
            success: function ($json) {
              init($chart, $json);
            }
          });
        }();

        // Methods
        function init($this, $json) {
          const chart = new Chart($this, $json);

          chart.options.tooltips.callbacks.title = function(tooltipItem, data) {
            return data.labels[tooltipItem[0].index];
          }

          chart.options.tooltips.callbacks.label = function(tooltipItem, data) {
            let dataset = data.datasets[tooltipItem.datasetIndex];
            let currentValue = dataset.data[tooltipItem.index];
            return '\\t' + dataset.label + ": " + currentValue;
          }

          // Save to jQuery object
          $this.data('chart', chart);
        }

      })();
    `
</script>
